lib/client.ex

```diff
@@ -0,0 +1,153 @@
+defmodule Kaper.Client do
+
+  defmacro __using__(config) do
+    quote do
+      @conf unquote(config)
+      def conf, do: @conf
+      def list_tasks(opts \\ %{}) do
+        unquote(__MODULE__).list_tasks(conf(), opts)
+      end
+      def define_task(type, dbrps, script, status \\ :enabled, vars \\ %{}, id \\ "", template_id \\ "") do
+        unquote(__MODULE__).define_task(conf(), type, dbrps, script, status, vars, id, template_id)
+      end
+      def get_task(id, dot_view \\ "attributes", script_format \\ "formatted") do
+        unquote(__MODULE__).get_task(conf(), id)
+      end
+      def delete_task(id) do
+        unquote(__MODULE__).delete_task(conf(), id)
+      end
+      def enable_task(id) do
+        unquote(__MODULE__).enable_task(conf(), id)
+      end
+      def disable_task(id) do
+        unquote(__MODULE__).disable_task(conf(), id)
+      end
+      def define_template(type, script, id \\ "") do
```

Comment:
It would probably help me to see an example but typically you won't want to have a using here, would be better to make them module functions, again will help to have docs.